File: JavaScript_Basic_04_2023_SolutionsOfProblems/02.Ex_PB-JS-Conditional-Statements/P04_ToyShop.js
function toyShop(input) {

    let tripCost = Number(input[0]);
    let puzzles = Number(input[1]);
    let dolls = Number(input[2]);
    let bears = Number(input[3]);
    let minions = Number(input[4]);
    let tracks = Number(input[5]);

    let profit = puzzles * 2.6 + dolls * 3 + bears * 4.1 + minions * 8.2 + tracks * 2;
    if (puzzles + dolls + bears + minions + tracks >= 50) {
        profit = profit * 0.75;
    }
    profit = profit * 0.9;
    if (profit < tripCost) {
        console.log(`Not enough money! ${(tripCost - profit).toFixed(2)} lv needed.`);
    } else {
        console.log(`Yes! ${(profit - tripCost).toFixed(2)} lv left.`);
    }
}

toyShop(["320",
    "8",
    "2",
    "5",
    "5",
    "1"])

File: Java_Basic_Jan_2023_SolutionsOfProblems/P02Ex_ConditionalStatements/P04_ToyShop.java
package P02Ex_ConditionalStatements;

import java.util.Scanner;

public class P04_ToyShop {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        double priceOfTrip = Double.parseDouble(scanner.nextLine());
        int puzzles = Integer.parseInt(scanner.nextLine());
        int dolls = Integer.parseInt(scanner.nextLine());
        int bears = Integer.parseInt(scanner.nextLine());
        int minions = Integer.parseInt(scanner.nextLine());
        int tracks = Integer.parseInt(scanner.nextLine());

        double amount = puzzles * 2.6 + dolls * 3 + bears * 4.1 + minions * 8.2 + tracks * 2;

        if (puzzles + dolls + bears + minions + tracks >= 50)
            amount = amount * 0.75;
        amount = amount * 0.9;

        if (amount >= priceOfTrip)
            System.out.printf("Yes! %.2f lv left.", amount - priceOfTrip);
        else
            System.out.printf("Not enough money! %.2f lv needed.", priceOfTrip - amount);

    }
}
